### ISSUE TITLE
Build error on FreeBSD related to PCK embedding

### ISSUE THREAD

**Issue by Calinou** [godotengine/godot#35100]
**Godot version:** Git https://github.com/godotengine/godot/commit/1d129f9becad39f60252a672eb2cf0e14056939d

**OS/device including version:** FreeBSD 12.0 (on Cirrus CI, see [this repository](https://github.com/Calinou/godot-ci-freebsd) for CI files)

**Issue description:** Building fails on FreeBSD with the following error:

```text
progress_finish(["progress_finish"], [])
Linking Static Library ==> modules/freetype/libfreetype_builtin.x11.opt.64.a
Ranlib Library         ==> modules/freetype/libfreetype_builtin.x11.opt.64.a
Linking Program        ==> bin/godot.x11.opt.64
/usr/bin/ld: error: platform/x11/pck_embed.ld:10: unknown directive: INSERT
>>> INSERT AFTER .rodata;
>>> ^
c++: error: linker command failed with exit code 1 (use -v to see invocation)
scons: *** [bin/godot.x11.opt.64] Error 1
scons: building terminated because of errors.
```

Maybe PCK embedding logic should be disabled on *BSD.

**Steps to reproduce:** Build Godot on FreeBSD using `tools=no target=release`.

**Comment by RandomShaper** [comment]
Which linker is used on this platform?

**Comment by akien-mga** [comment]
> Which linker is used on this platform?

It seems to be LLVM's `lld`: https://www.freebsd.org/cgi/man.cgi?ld(1)
Here's Calinou's Cirrus CI build log: https://cirrus-ci.com/task/6295550949588992

**Comment by akien-mga** [comment]
@RandomShaper As far as I know the linker script only works with `GNU ld` (`ld.bfd`) for now, so the simplest might be to parse `env["LD"] --version` and only enable it when it contains `GNU ld`:
```
$ ld.lld --version
LLD 8.0.0 (compatible with GNU linkers)

$ ld.gold --version
GNU gold (GNU Binutils 2.32) 1.16
Copyright (C) 2019 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.

$ ld.bfd --version
GNU ld (GNU Binutils) 2.32
Copyright (C) 2019 Free Software Foundation, Inc.
This program is free software; you may redistribute it under the terms of
the GNU General Public License version 3 or (at your option) a later version.
This program has absolutely no warranty.
```
(This is from Mageia 7 x86_64.)

**Comment by RandomShaper** [comment]
The "easy" way has got us so far. Let's consider this another known issue. I'll reimplement the feature so that the editor takes full care of adding the PCK section and at runtime it's searchedby parsing the file headers (the check for `GDPC` at the end, just in case, may be kept but it should fall back to the clever check if it fails).

That will fix issues with signing, resource editing and link problems.

**Comment by akien-mga** [comment]
While that's indeed the better fix, we still need Godot 3.2 to be able to build on FreeBSD, so this needs to be fixed (or worked around as I suggested) in coming days.

**Comment by RandomShaper** [comment]
Fair enough. I'll do it today.
